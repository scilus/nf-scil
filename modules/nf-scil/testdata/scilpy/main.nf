
process TESTDATA_SCILPY {
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
    val(archive)
    path(test_data_path)

    output:
    path("${archive.take(archive.lastIndexOf('.'))}") , emit: test_data_directory

    when:
    task.ext.when == null || task.ext.when

    exec:
    def remote = params.nf_scil_test_data_remote
    def database = params.nf_scil_test_database_path
    def data_associations = params.nf_scil_test_data_associations

    def data_home = file(
        System.getenv('XDG_DATA_HOME') ?: "${System.getenv('HOME')}/.local/share"
    )
    def storage = file("$data_home/nf-scil-test-archives")
    if ( !storage.exists() ) storage.mkdirs()

    def resource = file("$storage/${data_associations[archive]}")
    if ( !resource.exists() ) {
        def location = "${data_associations[archive][0..1]}/${data_associations[archive][2..-1]}"
        file("$remote/$database/$location").copyTo(resource)
    }

    def dest = task.workDir
    if ( !test_data_path.isEmpty() ) {
        dest = file("${task.workDir}/${test_data_path.getName()}")
        if ( !dest.exists() ) {
            test_data_path.mklink(dest)
        }
    }
    dest = dest.resolve("${archive.take(archive.lastIndexOf('.'))}")

    def content = new java.util.zip.ZipFile("$resource")
    content.entries().each{ entry ->
        def target = file("$dest/${entry.getName()}")
        if (entry.isDirectory()) {
            target.mkdirs();
        } else {
            target.getParent().mkdirs();
            file("$target").withOutputStream{ out -> out << content.getInputStream(entry)
            }
        }
    }
}
