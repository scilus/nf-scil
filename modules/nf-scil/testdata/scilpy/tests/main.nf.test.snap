{
    "load_bids - temp": {
        "content": [
            {
                "0": [
                    [
                        [
                            "result_complex_dwi_complex_rev_dwi.json:md5,8002324629222edb9217a77b27e60a34",
                            "result_complex_dwi_complex_rev_dwi_sbref.json:md5,720af6dfc6596814218a8e42d69b58c3",
                            "result_complex_dwi_complex_sbref.json:md5,60f3667851c3b6ed78f28a508a69341f",
                            "result_complex_dwi_epi.json:md5,60f3667851c3b6ed78f28a508a69341f",
                            "result_real_dwi_epi.json:md5,22adcf1437117b643cf292a82e7f7552",
                            "result_real_dwi_real_rev_dwi.json:md5,42cb893cc5185bb674a28a38f1e35a99",
                            "result_real_dwi_real_rev_dwi_sbref.json:md5,86343131de7e534024f0deb7697922c3",
                            "result_real_dwi_real_sbref.json:md5,22adcf1437117b643cf292a82e7f7552"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,5ccf597dc693f9e49c46da8fa7f6c839"
                ],
                "test_data_directory": [
                    [
                        [
                            "result_complex_dwi_complex_rev_dwi.json:md5,8002324629222edb9217a77b27e60a34",
                            "result_complex_dwi_complex_rev_dwi_sbref.json:md5,720af6dfc6596814218a8e42d69b58c3",
                            "result_complex_dwi_complex_sbref.json:md5,60f3667851c3b6ed78f28a508a69341f",
                            "result_complex_dwi_epi.json:md5,60f3667851c3b6ed78f28a508a69341f",
                            "result_real_dwi_epi.json:md5,22adcf1437117b643cf292a82e7f7552",
                            "result_real_dwi_real_rev_dwi.json:md5,42cb893cc5185bb674a28a38f1e35a99",
                            "result_real_dwi_real_rev_dwi_sbref.json:md5,86343131de7e534024f0deb7697922c3",
                            "result_real_dwi_real_sbref.json:md5,22adcf1437117b643cf292a82e7f7552"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5ccf597dc693f9e49c46da8fa7f6c839"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-26T14:26:21.40363"
    },
    "load_bids - work": {
        "content": [
            {
                "0": [
                    [
                        [
                            "result_complex_dwi_complex_rev_dwi.json:md5,8002324629222edb9217a77b27e60a34",
                            "result_complex_dwi_complex_rev_dwi_sbref.json:md5,720af6dfc6596814218a8e42d69b58c3",
                            "result_complex_dwi_complex_sbref.json:md5,60f3667851c3b6ed78f28a508a69341f",
                            "result_complex_dwi_epi.json:md5,60f3667851c3b6ed78f28a508a69341f",
                            "result_real_dwi_epi.json:md5,22adcf1437117b643cf292a82e7f7552",
                            "result_real_dwi_real_rev_dwi.json:md5,42cb893cc5185bb674a28a38f1e35a99",
                            "result_real_dwi_real_rev_dwi_sbref.json:md5,86343131de7e534024f0deb7697922c3",
                            "result_real_dwi_real_sbref.json:md5,22adcf1437117b643cf292a82e7f7552"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,5ccf597dc693f9e49c46da8fa7f6c839"
                ],
                "test_data_directory": [
                    [
                        [
                            "result_complex_dwi_complex_rev_dwi.json:md5,8002324629222edb9217a77b27e60a34",
                            "result_complex_dwi_complex_rev_dwi_sbref.json:md5,720af6dfc6596814218a8e42d69b58c3",
                            "result_complex_dwi_complex_sbref.json:md5,60f3667851c3b6ed78f28a508a69341f",
                            "result_complex_dwi_epi.json:md5,60f3667851c3b6ed78f28a508a69341f",
                            "result_real_dwi_epi.json:md5,22adcf1437117b643cf292a82e7f7552",
                            "result_real_dwi_real_rev_dwi.json:md5,42cb893cc5185bb674a28a38f1e35a99",
                            "result_real_dwi_real_rev_dwi_sbref.json:md5,86343131de7e534024f0deb7697922c3",
                            "result_real_dwi_real_sbref.json:md5,22adcf1437117b643cf292a82e7f7552"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5ccf597dc693f9e49c46da8fa7f6c839"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-26T14:26:28.259429"
    }
}